### DISCUSSION TITLE
Exception not detected when executing commands with SSHOperator

### DISCUSSION THREAD

**Discussion by Rodrigo507** [apache/airflow#22515] [Q&A]
Hi, I am using the SSHOperator to execute a Shell script (Execute a docker command), and it is running fine. But the problem I have is that when the container has failure (**Exception**) the Dag does not detect it and marks it as if the execution is ok.

In the .sh file I execute this line 
`sh -c "cd /home/***/***/example_airflow_app/app/ && /usr/local/bin/docker-compose run airflow_test" `

The container lifts perfectly, but when an Exception occurs, airflow does not detect it. 

`command=(
            '/home/***/***/example_airflow_app/app/source/file.sh 2>&1 '
        ) `

It shows me the error in the Task log, but as I repeat airflow does not take it as an error.

![image](https://user-images.githubusercontent.com/32920890/160015771-2cbe628a-6871-48f0-8833-6cac78332e3f.png)

**Comment by potiuk** [comment]
Likely the command does not exit with error. Make sure that the command you run exits with exit code == 0.

**Reply by potiuk** [reply]
There might be many reasons BTW. Maybe your shell script does not use `set -e` ? In this case it will exit with the exit code of the last instructions executed (which might be  a different instruction than you think

**Reply by potiuk** [reply]
This is how shell works BTW. Not Airflow.
